### PR TITLE
ceph-daemon: enable dashboard during bootstrap

### PR DESCRIPTION
**Pull request by liewegas**

### CODE REVIEW

File: src/ceph-daemon [hunk 1]
```diff
@@ -1096,6 +1100,25 @@ def command_bootstrap():
         logger.info('Adding host %s...' % host)
         cli(['orchestrator', 'host', 'add', host])
 
+    if not args.skip_dashboard:
+        logger.info('Enabling the dashboard module...')
+        cli(['mgr', 'module', 'enable', 'dashboard'])
+        logger.info('Generating a dashboard self-signed certificate...')
+        cli(['dashboard', 'create-self-signed-cert'])
```
Comment (sebastian-philipp) [inline src/ceph-daemon]: AFAIL, @tserong did run into a race here. I guess we have to wait, till the module is successfully enabled?

File: src/ceph-daemon [hunk 2]
```diff
@@ -809,6 +809,10 @@ def command_version():
 ##################################
 
 def command_bootstrap():
+    # some initial validation
+    if not args.skip_dashboard and not args.initial_dashboard_password:
+        raise RuntimeError('Must specify either --initial-dashboard-password <password> or --skip-dashboard')
+
     fsid = args.fsid or make_fsid()
     hostname = get_hostname()
```
Comment (capri1989) [inline src/ceph-daemon]: I just looked at the hostname function and shouldn't we call socket.getfqdn() first and only if the fqdn isn't set we could query socket.gethostname() ?

Comment (liewegas) [reply src/ceph-daemon]: for the dashboard URL, yes; for the mon name, i think we want the short name. will fix.

File: src/ceph-daemon (hunk 1 above)
Comment (liewegas) [reply src/ceph-daemon]: yep, fixed

**Comment by liewegas** [comment]
Hmm, one other question: should this generate a random password instead of prompting for one?

**Comment by capri1989** [comment]
> Hmm, one other question: should this generate a random password instead of prompting for one?

I think the perfect solution would be if you could configure it if you want to do so beforehand with something like PASSWORD=mypass and pass it as a command line option and if not a random password will be generated.

**Comment by liewegas** [comment]
> > Hmm, one other question: should this generate a random password instead of prompting for one?
> 
> I think the perfect solution would be if you could configure it if you want to do so beforehand with something like PASSWORD=mypass and pass it as a command line option and if not a random password will be generated.

updated!

**Comment by liewegas** [comment]
retest this please

**Comment by liewegas** [comment]
> LGTM. The one niggle I have is the text "Extracting ceph user uid/gid from container image"...on first run, this can take a while and may appear to hang. I'd update to say downloading/extracting so the user knows more about what's going on

We could add in an explicit ``podman pull`` step so that it's clear what the slow part is (and the uid/gid part will then always be fast)

**Pull request merged**